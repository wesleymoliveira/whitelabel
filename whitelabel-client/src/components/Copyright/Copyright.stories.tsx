import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { Copyright } from './Copyright';

export default {
	title: 'Copyright',
	component: Copyright
} as Meta;

export const Default: Story = () => <Copyright />;
