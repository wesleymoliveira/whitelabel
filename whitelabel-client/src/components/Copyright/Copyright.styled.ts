import styled from '@emotion/styled';

export const BottomGradient = styled.div`
	position: absolute;
	bottom: 0;
	width: 100vw;
	height: 200px;
	background: radial-gradient(
		102.86% 102.86% at 47.95% 160.94%,
		#ffc244 0%,
		rgba(255, 194, 68, 0) 100%
	);
`;
