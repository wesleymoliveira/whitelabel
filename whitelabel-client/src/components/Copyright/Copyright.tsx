import * as React from 'react';
import Typography from '@mui/material/Typography';
import MuiLink from '@mui/material/Link';
import Image from 'next/image';
import { Box } from '@mui/material';

export const Copyright = () => {
	return (
		<Box
			data-testid='copyright'
			sx={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}
		>
			<Typography fontFamily='Roboto, sans-serif' fontWeight={400} fontSize={14}>
				Powered by
			</Typography>
			<MuiLink color='inherit' href='/'>
				<Image
					src='https://res.cloudinary.com/dgop0ic8b/image/upload/v1650298129/icons/logo_vjopfc.svg'
					alt='sortium logo'
					height={35}
					width={150}
				/>
			</MuiLink>
		</Box>
	);
};
