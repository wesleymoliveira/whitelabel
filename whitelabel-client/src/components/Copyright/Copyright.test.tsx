import { render, screen } from '@testing-library/react';

import { Copyright } from './Copyright';

describe('<Copyright />', () => {
	it('should render the component', () => {
		const { container } = render(<Copyright />);

		expect(screen.getByTestId(/copyright/i)).toBeInTheDocument();

		expect(container.firstChild).toMatchSnapshot();
	});
});
