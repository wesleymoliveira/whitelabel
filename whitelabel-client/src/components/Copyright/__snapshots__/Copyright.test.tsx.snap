// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Copyright /> should render the component 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  font-size: 14px;
}

.emotion-1 {
  margin: 0;
  color: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: inherit;
}

.emotion-1:hover {
  text-decoration-color: inherit;
}

<div
  data-testid="copyright"
>
  <p
    class="MuiTypography-root MuiTypography-body1 emotion-0"
  >
    Powered by
  </p>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-1"
    href="/"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27150%27%20height=%2735%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="sortium logo"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </a>
</div>
`;
